scipy/stats/_discrete_distns.py

```diff
@@ -1528,6 +1529,152 @@ def _rvs(self, a, size=None, random_state=None):
                         name='dlaplace', longname='A discrete Laplacian')
 
 
+class poisson_binom_gen(rv_discrete):
+    r"""A Poisson Binomial discrete random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    binom
+
+    Notes
+    -----
+    The probability mass function for `poisson_binom` is:
+
+    .. math::
+
+     f(k; p_1, p_2, ..., p_n) = \sum_{A \in F_k} \prod_{i \in A} p_i \prod_{j \in A^C} 1 - p_j
+
+    where :math:`k \in \{1, 2, \dots, n-1, n\}`, :math:`F_k` is the set of all
+    subsets of :math:`k` integers that can be selected :math:`\{1, 2, \dots, n-1, n\}`,
+    and :math:`A^C` is the complement of a set :math:`A`.
+
+    `poisson_binom` takes a single array arguments for shape parameters :math:`p_i` where
+    the last axis corresponds with the index :math:`i` and any others are batch dimensions.
+
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] "Poisson binomial distribution", Wikipedia, https://en.wikipedia.org/wiki/Poisson_binomial_distribution
+
+    %(example)s
+
+    """
+    def _shape_info(self):
+        # message = 'Fitting is not implemented for this distribution."
+        # raise NotImplementedError(message)
+        return []
+
+    def _argcheck(self, *args):
+        conds = [(0 < arg) & (arg < 1) for arg in args]
+        conds = np.stack(conds)
+        return np.all(conds, axis=0)
+
+    def _rvs(self, *args, size=None, random_state=None):
```

Comment:
This currently doesn't follow the usual (but very confusing) convention by which `size` must be the full size of the output array, including the dimensions corresponding with the shape parameters.

Reply:
Err... now it does. The way `loc` is treated by the interface, it's tough to get around this.